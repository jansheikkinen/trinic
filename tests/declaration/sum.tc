// Sum

sum TestSum
  One(int8),
  Two
end

sum TestSum(T, V)
  One(T, V),
  Two(T),
  Three
end
