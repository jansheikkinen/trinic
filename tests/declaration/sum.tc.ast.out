
### AST GENERATION ###
Declaration SUM
  Arg SUMARG
    Sum_Arg One
      Arg SUMARGTYPE
        Sum_Arg_Type 
          Type TYPE INT8
    Sum_Arg Two
Declaration SUM
  Arg GENERIC
    IDENTIFIER T
    IDENTIFIER V
  Arg SUMARG
    Sum_Arg One
      Arg SUMARGTYPE
        Sum_Arg_Type T
        Sum_Arg_Type V
    Sum_Arg Two
      Arg SUMARGTYPE
        Sum_Arg_Type T
    Sum_Arg Three
